### PR TITLE
Squid queries

### PR DESCRIPTION
**Pull request by petebryan**
Fixes #
None

## Proposed Changes
New detections:
- squid proxy logs showing connections to ToR proxies
- squid proxy logs showing connections to cryptomining pools
New hunting queries:
- disabled accounts using squid proxy
- malformed squid proxy requests
- commonly abused TLDs in squid proxy logs
- traffic volume anomalies in squid proxy logs
New parser:
- Squid proxy syslog events

All of these items are predicated on squid logs being onboarded via Syslog as opposed to as a custom log source.

### CODE REVIEW

**Comment by oshezaf** [comment]
@petebryan : 
1- Squid had configurable logging format. I assume you use the default. Make sense to mention that in the description?
2- You may want to create a squid parser instead of putting the parser in every query.

**Comment by petebryan** [comment]
> @petebryan :
> 1- Squid had configurable logging format. I assume you use the default. Make sense to mention that in the description?
> 2- You may want to create a squid parser instead of putting the parser in every query.

1. Yep its the default format this is using as I wanted this to work with the largest number of squid set ups - I will add comments to this affect in descriptions.
2. I have done a separate parser but wanted to include them in each query so that if someone just takes one query on its own or does something like give the parser function a different name then it will still work.

**Comment by petebryan** [comment]
Is someone able to review and merge this PR please?

**Comment by shainw** [comment]
I will later today when I am in front of a computer.  Shain

Get Outlook for Android<https://aka.ms/ghei36>

________________________________

File: Detections/Syslog/squid_cryptomining_pools.txt [hunk 1]
```diff
@@ -0,0 +1,66 @@
+// Name: Squid proxy events for ToR proxies 
+//
+// Id: 90d3f6ec-80fb-48e0-9937-2c70c9df9bad
+//
+// Description: Check for squid proxy events associated with common ToR proxies. This query presumes the default squid log format is being used.
+//
+// DataSource: #Syslog
```
Comment (shainw) [inline Detections/Syslog/squid_cryptomining_pools.txt]: Minor change, but we want to go forward with a different format.  We want to include the DataConnector and the DataType, example:
DataConnector: #AzureActiveDirectory, #Office365; DataTypes: #SigninLogs, #OfficeActivity

For all of yours, let's go with - DataConnector: #Syslog; DataTypes: #Squid

File: Detections/Syslog/squid_tor_proxies.txt [hunk 2]
```diff
@@ -0,0 +1,106 @@
+// Name: Squid proxy events related to mining pools
+//
+// Id: 80733eb7-35b2-45b6-b2b8-3c51df258206
+// Description: check for squid proxy events associated with common mining pools .This query presumes the default squid log format is being used.
+//
+// DataSource: #Syslog
```
Comment (shainw) [inline Detections/Syslog/squid_tor_proxies.txt]: Let's go with - DataConnector: #Syslog; DataTypes: #Squid

File: Hunting Queries/Syslog/disabled_account_squid_usage.txt [hunk 3]
```diff
@@ -0,0 +1,35 @@
+// Name: Disabled accounts using squid proxy
+//
+// Id: 959fe0f0-7ac0-467c-944f-5b8c6fdc9e72
+//
+// Description: Look for accounts that have a been recorded as disabled by AD in the previous week but are still using the proxy during 
+// the current week. This query presumes the default squid log format is being used.
+//
+// DataSource: #Syslog
```
Comment (shainw) [inline Hunting Queries/Syslog/disabled_account_squid_usage.txt]: Let's go with - DataConnector: #Syslog; DataTypes: #Squid

File: Hunting Queries/Syslog/squid_abused_tlds.txt [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+// Id: 7aaa7675-1580-47d8-a404-039cb7284279
+//
+//
+// Description: some top level domains (TLDs) are more commonly associated with malware for a range of 
+// reasons - including how easy domains on these TLDs are to obtain. Many of these may be undesirable 
+// from an enterprise policy perspective. 
+// The clientCount column provides an initial insight into how widespread the domain usage is across the estate. 
+// This query presumes the default squid log format is being used.
+//
+// DataSource: #Syslog
```
Comment (shainw) [inline Hunting Queries/Syslog/squid_abused_tlds.txt]: Let's go with - DataConnector: #Syslog; DataTypes: #Squid

File: Hunting Queries/Syslog/squid_malformed_requests.txt [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+//
+// Id: edbeec9f-86b9-475d-8a42-cc7b95ad2baa
+//
+//
+// Description: Malformed web requests are sometimes used for Reconnaissance to detect the presence of network security devices.
+// Hunting for a large number of requests from a single source may assist in locating compromised hosts. Note: internal sites may
+// be detected by this query and may need excluding on a individual basis. This query presumes the default squid log format is
+// being used.
+//
+// DataSource: #Syslog
```
Comment (shainw) [inline Hunting Queries/Syslog/squid_malformed_requests.txt]: Let's go with - DataConnector: #Syslog; DataTypes: #Squid

File: Hunting Queries/Syslog/squid_volume_anomalies.txt [hunk 6]
```diff
@@ -0,0 +1,35 @@
+// Name: Squid data volume timeseries anomalies
+//
+// Id: e472c490-4792-4f12-8b6b-6ab3e0404d35
+//
+//
+// Description: Malware infections or data exfiltration activity often leads to anomalies in network data volume
+// this hunting query looks for anomalies in the volume of bytes traversing a squid proxy. Anomalies require further 
+// investigation to determine cause. This query presumes the default squid log format is being used.
+//
+// DataSource: #Syslog
```
Comment (shainw) [inline Hunting Queries/Syslog/squid_volume_anomalies.txt]: Let's go with - DataConnector: #Syslog; DataTypes: #Squid

**Comment by petebryan** [comment]
Thanks @shainw - I've updated all the DataConnector stings to the correct format now.

**Pull request merged**